js/app/session.js

```diff
@@ -35,6 +35,18 @@ define(['userData', 'login'], function(userData, login) {
 		return false;
 	}
 
+	function theWordsFit(from, to) {
+		// test both cases, because user may reverse the words
+		ctxWords.font = WORD_FONT;
+		if (isWordFittingTheScreen(from, MAX_WORD_LENGTH) && isWordFittingTheScreen(to, MAX_WORD_LENGTH)) {

```

Comment:
This is better than it was if I remember correctly!

The most elegant solution is to take the font that you consider when measuring the font as an argument to isWordFittingTheScreen. and to change the font inside that function. 

Then the code would look like this: 

return 
  (isWordFittingTheScreen(from, MAX_WORD_LENGTH, WORD_FONT) &&   
  isWordFittingTheScreen(to,MAX_WORD_LENGTH, WORD_FONT && 
  isWordFittingTheScreen(to,MAX_TRANSLATION_LENGTH, TRANSLATION_FONT && 
  isWordFittingTheScreen(to,MAX_TRANSLATION_LENGTH, TRANSLATION_FONT) 

Am I right?